cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -1744,6 +1746,19 @@ impl ManagedEnvironment {
                 }))?;
             }
         }
+
+        // check if there are any changes to push by comparing branch hashes
+        let project_branch_commit = self
+            .floxmeta
+            .git
+            .branch_hash(&project_branch)
+            .map_err(ManagedEnvironmentError::GitBranchHash)?;
+        let sync_branch_commit = self.floxmeta.git.branch_hash(&sync_branch).ok();
+
+        if Some(project_branch_commit.clone()) == sync_branch_commit {
+            return Err(ManagedEnvironmentError::NothingToPush);
+        }
+
```

Comment:
non-blocking: last time around someone approached this issue, it was suggested that we could just let git do this, i.e. run `git push --porcelain` and parse its output: https://github.com/flox/flox/pull/2927#discussion_r2089449060